### PR TITLE
[WG Platforms] Rewrite section on metrics

### PR DESCRIPTION
**Pull request by joshgav**
This is a rewrite of the section on metrics and measures based in part on discussions in #278. It categorizes meters an organization might track into the following three areas then provides some typical examples for each.

1. User satisfaction and productivity
1. User autonomy and process automation
1. Product and feature delivery

Here's the rendered version for easier review: <https://github.com/joshgav/tag-app-delivery/blob/0771a691c4a23b6cc00c1558efb6df3c8d58132c/platforms-whitepaper/v1alpha1/paper.md#how-to-measure-the-success-of-platforms>

@Saim-Safdar WDYT?

### CODE REVIEW

**Review by Saim-Safdar — COMMENTED** [review]
Those are much-needed metrics for consideration, but I believe we need to highlight Outputs vs outcomes vs impact, thinking here is some teams don't use a metrics-driven approach instead using tracking and reporting. This could be caused by a lack of product thinking in platform teams or technical complexity in extracting the data (or both).

Setting an outcome-based goal (i.e. increasing the number of weekly querying users) for a specific quarter and expecting the team to deliver the outputs.

Another way to look into it is, the business KPIs your company sets won’t work immediately for your platform team. You’ll therefore need to find your own platform team KPIs, are we planning to cover or are we providing some insights into how to internalize and externalize KPIs for the platform team as the business KPI's changing or are these two separate things?

**Comment by joshgav** [comment]
Thanks @Saim-Safdar. I hear you on considering objectives or outcomes alongside metrics. Do you think the Attributes section describes these higher order objectives? Maybe we can make it clear that the measures in this section represent attainment of the objectives described in the Attributes and Why sections. WDYT?

Thanks for bringing up business metrics versus platform metrics. The text proposed here nods to that in saying things like “the success of the platform depends on the success of its users”. Is there another contrast between platform and product objectives and metrics we could add? Thanks!

**Comment by joshgav** [comment]
Hi @Saim-Safdar - I took another pass, focusing particularly on aligning measures with objectives (the attributes described earlier in the paper). I also chatted with @abangser and got some valuable tips and feedback. PTAL!

Reading your earlier comment again, you may be saying we should recommend enterprises or platform teams set quantitative outcome targets like "NPS > 50" or "MAU > 2000" rather than just tracking the metrics. If so, we could merge this as is yet still add that idea later.

**Review by abangser — APPROVED** [review]
Love it @joshgav ! In particular love that you one again point towards thinnest viable while also providing next step options.

File: platforms-whitepaper/v1alpha1/paper.md [hunk 1]
```diff
@@ -306,32 +306,63 @@ the platform team include the following:
...
-- Time for a new contributor to submit their first (or 10th!) PR
-- Number of contributions from users, e.g. Backstage plugins, Crossplane
-  compositions
-- Number of projects which diverge from "golden paths" and provided capabilities
-- Number of tickets opened related to platform capabilities
-- Developer satisfaction in the enterprise
-    - Net Promoter Score (NPS)
-    - SPACE framework: https://queue.acm.org/detail.cfm?id=3454124
+Enterprises will want to measure whether their platform initiatives are
+delivering their anticipated values. Also, throughout this paper we've
```
Comment (abhinavm-rafay) [inline platforms-whitepaper/v1alpha1/paper.md:310]: this is a little vague @joshgav can we clarify what we mean here by anticipated values?

File: platforms-whitepaper/v1alpha1/paper.md [hunk 2]
```diff
@@ -306,32 +306,63 @@ the platform team include the following:
...
+emphasized the importance of treating internal platforms as products, and good
+product management depends on quantitative and qualitative measurement of a
+product's performance. To meet these requirements, internal platform teams
+should gather user feedback and measure user activities.
```
Comment (abhinavm-rafay) [inline platforms-whitepaper/v1alpha1/paper.md:314]: should continuously gather

File: platforms-whitepaper/v1alpha1/paper.md [hunk 3]
```diff
@@ -306,32 +306,63 @@ the platform team include the following:
...
+
+As with other aspects of internal platforms, though, platform teams should use
+the smallest viable effort to gather the feedback they need. We'll suggest
+metrics here but simple surveys or question and answer sessions may be most
```
Comment (abhinavm-rafay) [inline platforms-whitepaper/v1alpha1/paper.md:318]: one aspect here to talk about is also observing how users use your platforms - ethnographic studies these kind of studies can prove to be super useful

File: platforms-whitepaper/v1alpha1/paper.md [hunk 4]
```diff
@@ -306,32 +306,63 @@ the platform team include the following:
...
+Categories of metrics that will help enterprises and platform teams understand
+the impact of their platforms include the following:
+
+1. User satisfaction and productivity
+
+The first quality sought by many platforms is to improve user experience in order
+to increase productivity. Metrics that reflect user satisfaction and
+productivity include the following:
+
+- Active users and resources: number of capabilities provisioned and used from the platform
```
Comment (abhinavm-rafay) [inline platforms-whitepaper/v1alpha1/paper.md:330]: should consider metrics here like DAU and MAU or monthly adoption rate as cited in the Palo Alto networks example

File: platforms-whitepaper/v1alpha1/paper.md [hunk 5]
```diff
@@ -306,32 +306,63 @@ the platform team include the following:
...
+
+1. Product and feature delivery
+
+The ultimate objective of internal platforms is to deliver business value to
+customers faster, so measuring impact on a business's own product and feature
+releases demonstrates that the objectives of the platform are being met. The
+DevOps Research and Assessment (DORA) institute at Google suggests [[4]]
+tracking the following metrics:
+
+- Deployment frequency
```
Comment (abhinavm-rafay) [inline platforms-whitepaper/v1alpha1/paper.md:354]: here if we want to be more specific we can give examples such as 

- decrease in time from commit to deploy
- decrease in MTTR 
- decrease in change fail rate 
- increase in deployments

File: platforms-whitepaper/v1alpha1/paper.md (hunk 1 above)
Comment (joshgav) [reply platforms-whitepaper/v1alpha1/paper.md:310]: I hear you! Yes, I'll change to "whether their platform initiatives are delivering the values and attributes discussed above". I'm thinking the Attributes section has the Objectives that inform the Key Results in this section.

File: platforms-whitepaper/v1alpha1/paper.md (hunk 3 above)
Comment (joshgav) [reply platforms-whitepaper/v1alpha1/paper.md:318]: Yes, I hear you. Changing to "We'll suggest metrics here but simple surveys and analysis of user behavior may be most valuable initially."

File: platforms-whitepaper/v1alpha1/paper.md (hunk 5 above)
Comment (joshgav) [reply platforms-whitepaper/v1alpha1/paper.md:354]: Well you won't know if they increase or decrease till you measure them!

**Comment by joshgav** [comment]
Thanks for the feedback everyone! Still open to further iteration!

**Pull request merged**